e2e/utils/simulator/CreateSimulationDriver.ts

```diff
@@ -0,0 +1,205 @@
+import { expect, Page } from '@playwright/test';
+import {
+  assertDebugToken,
+  assertOverlappingTestCase,
+  assertRecurringTestCase,
+  getRecurringSettings,
+  screenshotPath,
+} from './utils';
+import {
+  RangeOrder,
+  debugTokens,
+  CreateStrategyTestCase,
+  Direction,
+  Setting,
+  StrategyType,
+} from './types';
+import { waitModalClose, waitModalOpen, waitTooltipsClose } from '../modal';
+import { screenshot, shouldTakeScreenshot, waitFor } from '../operators';
+import { MainMenuDriver } from '../MainMenuDriver';
+import { dayjs } from '../../../src/libs/dayjs';
+
+export class CreateSimulationDriver {
+  constructor(private page: Page, private testCase: CreateStrategyTestCase) {}
+
+  async waitForPriceChart(timeout?: number) {
+    await waitFor(this.page, 'price-chart', timeout);
+  }
+
+  async screenshotPriceChart() {
+    const priceChart = this.page.getByTestId('price-chart');
+    await priceChart.scrollIntoViewIfNeeded();
+    await screenshot(
+      priceChart,
+      screenshotPath(this.testCase, 'simulator-input-price')
+    );
+  }
+
+  async waitForDisclaimerModal(timeout?: number) {
+    await waitFor(this.page, 'sim-disclaimer-modal', timeout);
+  }
+
+  getForm() {
+    return this.page.getByTestId('create-simulation-form');
+  }
+
+  getFormSection(direction: Direction) {
+    const form = this.page.getByTestId(`${direction}-section`);
+    return {
+      locator: form,
+      price: () => form.getByTestId('input-price'),
+      min: () => form.getByTestId('input-min'),
+      max: () => form.getByTestId('input-max'),
+      budget: () => form.getByTestId('input-budget'),
+      setting: (setting: 'limit' | 'range') => {
+        return form.getByTestId(`tab-${setting}`);
+      },
+    };
+  }
+
+  async selectMonthYear(monthYear: string) {
+    const findMonthYear = async () => {
+      const isMonthYearVisible = await this.page.isVisible(
+        `text='${monthYear}'`
+      );
+      if (isMonthYearVisible) {
+        return;
+      } else {
+        await this.page.getByTestId('date-picker-left-arrow').click();
+        return await findMonthYear();
+      }
+    };
+    await findMonthYear();
+  }
+
+  async selectDay(day: string, monthYear: string) {
+    const dayButton = this.page
+      .getByLabel(monthYear)
+      .getByText(day, { exact: true });
+    const dayButtonCount = await dayButton.count();
+    // If the month has more than one day with the same number, find right one to press
+    if (dayButtonCount > 1) {
+      if (Number(day) < 15) {
+        await dayButton.nth(0).click();
+      } else {
+        await dayButton.nth(1).click();
+      }
+    } else {
+      await dayButton.click();
+    }
```

Comment:
Avoid nested when possible : 
```suggestion
    if (dayButtonCount > 1 && Number(day) < 15) {
      const index = Number(day) < 15 ? 0 : 1;
      await dayButton.nth(index).click();
    } else {
      await dayButton.click();
    }
```